src/FsAutoComplete.Core/Utils.fs

```diff
@@ -1,549 +1,603 @@
+    let splitAtChar (splitter: char) (s: string) =
+        match s.IndexOf splitter with
+        | -1 -> NoMatch
+        | n -> Split(s.[0..n-1], s.Substring(n+1))
+
+type ConcurrentDictionary<'key, 'value> with
+    member x.TryFind key =
+        match x.TryGetValue key with
+        | true, value -> Some value
+        | _ -> None
+
+type Path with
+    static member GetFullPathSafe (path: string) =
+        try Path.GetFullPath path
+        with _ -> path
+
+    static member GetFileNameSafe (path: string) =
+        try Path.GetFileName path
+        with _ -> path
+
+    /// Algorithm from https://stackoverflow.com/a/35734486/433393 for converting file paths to uris,
+    /// modified slightly to not rely on the System.Path members because they vary per-platform
+    static member FilePathToUri (filePath: string): string =
+        let filePath, finished =
+            if filePath.Contains "Untitled-" then
+                let rg = System.Text.RegularExpressions.Regex.Match(filePath, @"(Untitled-\d+).fsx")
+                if rg.Success then
+                    rg.Groups.[1].Value, true
+                else
+                    filePath, false
+            else
+                filePath, false
+
+        if not finished then
+            let uri = System.Text.StringBuilder(filePath.Length)
+            for c in filePath do
+                if (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9') ||
+                    c = '+' || c = '/' || c = '.' || c = '-' || c = '_' || c = '~' ||
+                    c > '\xFF' then
+                    uri.Append(c) |> ignore
+                // handle windows path separator chars.
+                // we _would_ use Path.DirectorySeparator/AltDirectorySeparator, but those vary per-platform and we want this
+                // logic to work cross-platform (for tests)
+                else if c = '\\' then
+                    uri.Append('/') |> ignore
+                else
+                    uri.Append('%') |> ignore
+                    uri.Append((int c).ToString("X2")) |> ignore
+
+            if uri.Length >= 2 && uri.[0] = '/' && uri.[1] = '/' then // UNC path
+                "file:" + uri.ToString()
+            else
+                "file:///" + (uri.ToString()).TrimStart('/')
+        else
+            "untitled:" + filePath
+
+    /// handles unifying the local-path logic for windows and non-windows paths,
+    /// without doing a check based on what the current system's OS is.
+    static member FileUriToLocalPath (uriString: string) =
+        /// a test that checks if the start of the line is a windows-style drive string, for example
+        /// /d:, /c:, /z:, etc.
+        let isWindowsStyleDriveLetterMatch (s: string) =
+            match s.[0..2].ToCharArray() with
+            | [| |]
+            | [| _ |]
+            | [| _; _ |] -> false
+            // 26 windows drive letters allowed, only
+            | [| '/'; c; ':' |] when Char.IsLetter c -> true
```

Comment:
Super minor nit: I know this code was just a move but I got confused when reading it.  Maybe rename `c` to `driverLetter`.